src/regressionmodel.jl

```diff
@@ -110,6 +129,36 @@ function effects!(reference_grid::DataFrame, model::RegressionModel;
     # return (; reference_grid..., depvar => eff, err_col => err)
 end
 
+# TODO: support the transformation method
+# in addition to the difference method
+# xref https://github.com/JuliaStats/GLM.jl/blob/c13577eaf3f418c58020534dd407532ee57f219b/src/glmfit.jl#L773-L783
+
+_invlink_and_deriv(invlink, η) = (invlink(η), ForwardDiff.derivative(invlink, η))
+_invlink_and_deriv(::typeof(identity), η) = (η, 1)
+# this isn't the best name because it sometimes returns the inverse link and sometimes the link (Link())
+# for now, this is private API, but we should see how this goes and whether we can make it public API
+# so local extensions (instead of Package-Extensions) are better supported 
+_model_link(::RegressionModel, invlink::Function) = invlink
+function _model_link(::RegressionModel, ::AutoInvLink)
+    @static if VERSION < v"1.9"
+        @error "AutoInvLink requires extensions and is thus not available on Julia < 1.9."
+    end
+    throw(ArgumentError("No appropriate extension is loaded for automatic " *
+                        "determination of the inverse link for this model type"))
```

Comment:
```suggestion
function _model_link(model::RegressionModel, ::AutoInvLink)
    msg = string("cannot automatically determine inverse link for models ",
                 "of type ", typeof(model))
    @static if isdefined(Base, :get_extension)
        msg *= "; no appropriate extension has been loaded"
    end
    throw(ArgumentError(msg))
```
It's not actually true that `AutoInvLink` is only available on Julia 1.9+ since it's defined and exported on all versions and its functionality can be made to work via explicit method extension.